plugins/agents/filesystem/Filesystem.go

```diff
@@ -0,0 +1,105 @@
+package filesystem
+
+import (
+	"io/ioutil"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"github.com/gansoi/gansoi/plugins"
+	"github.com/gansoi/gansoi/transports"
+)
+
+func init() {
+	plugins.RegisterAgent("filesystem", Filesystem{})
+}
+
+// Filesystem will check the condition of mounted storage devices.
+type Filesystem struct {
+	CommaSeparatedExcludedDevices string `json:"excludedDevices" description:"Comma-separated list of devices, that will we excluded from the check"`
+	excludedDevices               []string
+	parser                        commandParser
+}
+
+// RemoteCheck implements plugins.RemoteAgent.
+func (fs *Filesystem) RemoteCheck(transport transports.Transport, result plugins.AgentResult) error {
+	if fs.parser == nil {
```

Comment:
This will always be `nil`. But fear not, object creation in Go is fast.

Reply:
It will be nill, unless some sneaky test makes it a mock ;P

Reply:
Uhh, sneaky! Add a comment in the struct, and it should be obvious even for me ;)